### DISCUSSION TITLE
Support for preact

### DISCUSSION THREAD

**Discussion by IshanKBG** [markdoc/markdoc#309] [Q&A] [answered]
It would be awesome if there would be support for preact for rendering markdoc

**Comment by segphault** [comment, marked as answer]
Markdoc's React renderer actually supports Preact out of the box. The second parameter of the `Markdoc.renderers.react` function is typically the `React` object, but all it needs is an object with `createElement` and `Fragment` functions that match the React signatures. Preact includes that in its top-level exports, so you can do this:

```
import {render, createElement, Fragment} from 'preact';
import Markdoc from '@markdoc/markdoc';

const content = `# This is a test: {% $foo %}`;
const parsed = Markdoc.parse(content);
const transformed = Markdoc.transform(parsed, {
  variables: {
    foo: 'bar',
  },
});

const output = Markdoc.renderers.react(transformed, {createElement, Fragment});
render(output, document.body);
```
